lib/ReactViews/Analytics/InvokeFunction.jsx

```diff
@@ -32,6 +32,10 @@ const InvokeFunction = React.createClass({
         this.initializeParameters(nextProps);
     },
 
+    componentWillUpdate(nextProps, nextState) {
+        this.initializeParameters(nextProps);
+    },
+
```

Comment:
So, previously initializeParameters was only called in componentWillMount. But this got called the first time with 0 parameters, then render() would get called with 0 parameters. Then render() would get called again, this time with 3 parameters, but they would not have been initialised because componentWillMount is only be called before the first render(). I was looking for a way to call initalizeParameters before render() that triggered the first time. I agree this doesn't sound quite right though. Do you have a different idea of how I could fix this?